src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensHelpers.cs

```diff
@@ -149,9 +150,8 @@ static SemanticTokensHelpers()
             var isFinalized = document.Project.TryGetCompilation(out var compilation) && compilation == semanticModel.Compilation;
             document = frozenDocument;
 
-            var options = ClassificationOptions.From(document.Project);
-            var classifiedSpans = Classifier.GetClassifiedSpans(document.Project.Solution.Workspace.Services, semanticModel, textSpan, options, cancellationToken);
-            Contract.ThrowIfNull(classifiedSpans, "classifiedSpans is null");
+            using var _ = ArrayBuilder<ClassifiedSpan>.GetInstance(out var classifiedSpans);
+            await GetClassifiedSpansForDocumentAsync(document, textSpan, classifiedSpans, isRazorDoc, cancellationToken).ConfigureAwait(false);
```

Comment:
Not a fan of this kind of "secret return" inside of the `classifiedSpans`. Could `GetClassifiedSpansForDocumentAsync` return `Task<ClassifiedSpan[]>` so we can be explicit about what had work done on it? Also, if `ArrayBuilder<ClassifiedSpan>` is static couldn't we avoid passing it and just create the instance inside of `GetClassifiedSpansForDocumentAsync`?

Reply:
note: in terms of naming, we try to do 'Get' to have a return value, and 'Add' if someone will pass in the collection to fill.